prompting/base/miner.py

```diff
@@ -134,7 +132,7 @@ def run(self):
         except Exception as err:
             bt.logging.error("Error during mining", str(err))
             bt.logging.debug(print_exception(type(err), err, err.__traceback__))
-            self.should_exit = True
+            self.should_exit: bool = True
```

Comment:
We should avoid type hints for in-place variables, as they are generally clear to both static type checkers and readers without them
```suggestion
            self.should_exit = True
```